ports/stm32/timer.c

```diff
@@ -546,25 +577,28 @@ STATIC mp_obj_t pyb_timer_init_helper(pyb_timer_obj_t *self, size_t n_args, cons
     // set the TIM configuration values
     TIM_Base_InitTypeDef *init = &self->tim.Init;
 
-    if (args[0].u_obj != mp_const_none) {
+    if (args[ARG_freq].u_obj != mp_const_none) {
         // set prescaler and period from desired frequency
-        init->Prescaler = compute_prescaler_period_from_freq(self, args[0].u_obj, &init->Period);
-    } else if (args[1].u_int != 0xffffffff && args[2].u_int != 0xffffffff) {
+        init->Prescaler = compute_prescaler_period_from_freq(self, args[ARG_freq].u_obj, &init->Period);
+    } else if (args[ARG_period].u_int != 0xffffffff && args[ARG_period_hz].u_int != 0xffffffff) {
+        // set prescaler and period from desired period and scaler
+        init->Prescaler = compute_prescaler_period_from_t(self, args[ARG_period].u_int, args[ARG_period_hz].u_int, &init->Period);
+    } else if (args[ARG_prescaler].u_int != 0xffffffff && args[ARG_period].u_int != 0xffffffff) {
         // set prescaler and period directly
```

Comment:
If you switched the order of the two checks above (i.e. check for `prescaler` before `period_hz`) then you could set the default value of `period_hz` to 1000 and get some convergence towards the timer API used on the ESP32 and ESP8266 without breaking existing code.

Reply:
In this case I'd rather not... I think it'd make more sense for period to default to seconds (so `period_hz=1`) and maybe one day that can be done.

Reply:
Yes, it would certainly make more sense for `period_hz` to default to 1 but of course for that to be useful we would then want to put the floating point support back in, since most of the time people are going to need more resolution than one second, and I just took that out on the ESP8266 proposal!

When it comes to resolving the discussion in your RFC #2971, there are at least two problems: the APIs for timers are not always easy to use and the APIs are inconsistent across devices. I'd argue that even if a default of 1000 is not perfect it helps a little with the former and a lot with the later. I would rather see no default at all than a default to something different than on other platforms. Then again setting a default would only add new functionality and not break old code, and IMHO moving towards consistency would be a good thing.

Reply:
> I would rather see no default at all than a default to something different than on other platforms.

Agree.  That's why I left it as having no default in this initial implementation.

> I'd argue that even if a default of 1000 is not perfect it helps a little with the former and a lot with the later

Good points.  Well, one argument for making it milliseconds by default (not seconds) is that some things like poll.poll use milliseconds as their units.

Reply:
Milliseconds seem to be the default in many places. I think that on platforms that only have integers they are the most useful SI unit, since seconds provide insufficient resolution and microseconds provide insufficient range.